### DISCUSSION TITLE
Change device in the "Preview" mode

### DISCUSSION THREAD

**Discussion by RostyslavMoroziuk** [GrapesJS/grapesjs#4959] [Q&A] [answered]
I have custom buttons to change a device. We call `gjsEditor?.setDevice()` when the user clicks these custom device buttons.

If preview mode is activated then components become editable once I change the device.

Is it possible to change a device in preview mode and don't make components editable?

**Comment by artf** [comment, marked as answer]
There is a default command for selecting components which is turning on/off when you enable/disable the preview (same happening when you update the device). You can disable this default command manually once you've changed the device `editor.getModel().stopDefault()`
